src/cace/ari/access.h

```diff
@@ -132,6 +132,18 @@ struct ari_rptset_s *ari_init_rptset(ari_t *ari);
  */
 void ari_set_objref_path_textid(ari_t *ari, const char *ns_id, ari_type_t type_id, const char *obj_id);
 
+/** Set the ARI as an object reference with a specific text-named path.
+ *
+ * @param[in,out] ari The ARI value to modify.
+ * @param[in] has_ns Determine whether namespace path segment is present.
+ * @param[in] ns_id The namespace path segment.
+ * @param[in] has_type Determine whether object type path segment is present.
+ * @param type_id The object type path segment.
+ * @param[in] has_obj Determine whether object ID path segment is present.
+ * @param[in] obj_id The object ID path segment.
+ */
+void ari_set_objref_path_textid_opt(ari_t *ari, bool has_ns, const char *ns_id, bool has_type, ari_type_t type_id, bool has_obj, const char *obj_id);
```

Comment:
Consider using `const type *` pointers for optional parameters and allowing NULL pointer to indicate lack-of-value. The API doc for the parameter should indicate something like "The pointed-to lifetime does not need to outlast this function call." I can submit a patch commit to give an example if that helps.

Reply:
@BrianSipos Please see my most recent changes. For this one and below.